### PR TITLE
Fix service plan not loading when no overrides present

### PR DESCRIPTION
**Pull request by powerpbx**
Service plans do not load when there are no overrides present.  I believe this will fix that.

### CODE REVIEW

**Comment by lazedo** [comment]
@powerpbx thanks for the pr. it seems that the issue will be better handled at https://github.com/2600hz/kazoo/blob/master/core/kazoo_stdlib/src/kz_json.erl#L296
i'll let @jamesaimonetti figure out.

**Comment by jamesaimonetti** [comment]
@lazedo, @powerpbx and I talked briefly on IRC about this. My thought was:

Undoing the change where merging the empty object onto a non-empty object removes the non-empty object properties (this would make it more like merge_recursive/2 in that case). Tests would need to be refactored to support this assumption. This was more invasive and had wider reaching consequences.

For now, ignoring empty objects as a no-op gets things going they way they should for service plan overrides. So I'm fine with this PR to unblock the issue with empty overrides, with the thought that we reevaluate `kz_json:merge/2` and empty object handling.

File: core/kazoo_services/src/kz_service_plans.erl [hunk 1]
```diff
@@ -236,8 +238,9 @@ get_plan(PlanId, ResellerId, Services, ServicePlans) ->
 maybe_fetch_vendor_plan(PlanId, VendorId, VendorId, Overrides) ->
     case kz_service_plan:fetch(PlanId, VendorId) of
         'undefined' -> 'undefined';
-        ServicePlan ->
-            kzd_service_plan:merge_overrides(ServicePlan, Overrides)
+        ServicePlan when Overrides /= ?EMPTY_JSON_OBJECT ->
```
Comment (jamesaimonetti) [inline core/kazoo_services/src/kz_service_plans.erl]: My only comment is the project favors using `=/=` and `=:=` for equality tests.

**Comment by fenollp** [comment]
Sorry to jump on this a bit late: to me it would be cleaner to have 

```erlang
OverridesAreEmpty = kz_json:is_empty(Overrides),
...
+        ServicePlan when OverridesAreEmpty ->
```

and not include the json header file altogether.

**Comment by powerpbx** [comment]
What is the status of this? I tested my suggested fix and know it works.  I don't have my dev environment set up anymore so I am not in a position to test other peoples fixes.  It doesn't matter to me how it's done as long as it gets fixed.

What is the etiquette here when other people suggest something different?

**Comment by lazedo** [comment]
@powerpbx request was made to have the changes suggested by @fenollp commited instead. it does not rely on macros defined elsewhere. if the json library changes, then this would have to change also.

**Comment by powerpbx** [comment]
Do I need to do anything?

**Comment by fenollp** [comment]
This needs rebase for circleci, sorry...

**Comment by powerpbx** [comment]
I don't know what that means.

**Comment by fenollp** [comment]
I can also do this patch myself, will probably backport it too. Is this fine with you?

**Comment by powerpbx** [comment]
Hi,

Whatever you guys decide is fine with me.  Circleci is showing passed now.

**Comment by lazedo** [comment]
@powerpbx thanks for spotting and for your effort on fixing this. we merged @fenollp PR's  for master and 4.0 because of rebase issues with your branch.

**Pull request closed without merging**